### PR TITLE
FEPointEvaluation: Implement n_active_entries_per_quadrature_batch

### PR DESCRIPTION
**Pull request by bergbauer**
This function allows to only loop over active lanes in a quadrature batch.

FYI @ritthaler

### CODE REVIEW

File: include/deal.II/matrix_free/fe_point_evaluation.h [hunk 1]
```diff
@@ -2444,6 +2450,22 @@ inline typename FEPointEvaluationBase<n_components_, dim, spacedim, Number>::
...
+template <int n_components_, int dim, int spacedim, typename Number>
+unsigned int
+FEPointEvaluationBase<n_components_, dim, spacedim, Number>::
+  n_active_entries_per_quadrature_batch(unsigned int q)
+{
+  Assert(stride == 1,
+         ExcMessage(
+           "Calling this function only makes sense in fully vectorized mode."));
+  if (q == n_q_batches - 1)
+    return n_q_points_scalar & (n_lanes_user_interface - 1);
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_point_evaluation.h:2462]: Is there a reason you change from the modulo operation to this form? In my experience, compilers reduce `module (power of two)` into `bitwise and with (power of two - 1)`.

Comment (bergbauer) [reply include/deal.II/matrix_free/fe_point_evaluation.h]: We also use it here https://github.com/dealii/dealii/blob/31a1263477fc0d5b85bb7a2456ebb32744616593/include/deal.II/matrix_free/fe_point_evaluation.h#L2833-L2835

I can change this to modulo if you prefer.

Comment (kronbichler) [reply include/deal.II/matrix_free/fe_point_evaluation.h]: I don't care too much, I was just wondering why you changed it. I think it is good to be consistent within the code base, and I believe the modulo is used more frequently as I see from a quick search (but it is hard to filter out other uses of `&`).

**Pull request merged**